hazelcast/src/main/java/com/hazelcast/internal/blackbox/Blackbox.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.blackbox;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * The Blackbox is responsible for recording all kinds of Hazelcast/JVM specific information to
+ * help out with all kinds of issues.
+ *
+ * Each HazelcastInstance has its own Blackbox.
+ *
+ * A sensor for the Blackbox is build up of the following parts:
+ * <ol>
+ * <li>Parameter: uniquely identifies a Sensor</li>
+ * <li>SensorInput: responsible for providing a value. The Blackbox can sample these inputs</li>
+ * <li>SensorView: responsible for providing a view on the sensor measure content</li>
+ * </ol>
+ *
+ * It is possible to get a view on a non existing sensor; then a sensor is created but no input is provided.
+ * The reason why this is done is that not every SensorInput always is available. E.g. if we have an
+ * operation.count sensor, then we want to be able to read it. But if a new OperationService implementation is
+ * made that magically can't provide this sensor, than nothing should break.
+ *
+ * The Blackbox is also lenient regarding duplicate registrations of sensors. So if a Sensor is created and
+ * an SensorInput is set and a new registration for the same Sensor is done, the old SensorInput is overwritten.
+ * The reason to be lenient is that the Blackbox should not throw exception. Of course there will be a log
+ * warning.
+ *
+ * TODO:
+ * - uptime
+ * - number of invocations
+ * - number of failed invocation
+ * - number of retried invocations
+ * - back pressure information
+ * - runtime.availableMemory=NA  <<=== sensor doesn't exist.
+ *
+ * - tcp total number of created client connections
+ * - cluster: number of heart beats send
+ * - cluster: number of heart beats received
+ * - io number of rebalances
+ * - number per type of packet send
+ * - number per type of packet received
+ */
+public interface Blackbox {
+
+    /**
+     * Registers an object by scanning for the Sensor input annotations.
+     *
+     * If the same object already is registered, nothing bad happens.
+     *
+     * If an object has no sensor annotations, the call is ignored.
```

Comment:
The problem is that a single object can have multiple fields. So what does true or false mean?

For the time being the current solution is good enough. But I 3.6 I want to revert back to a fail fast approach. So sensors need to be deregistered for new sensors can be registered. The primary reason to go for this lenient approach is that I don't want exceptions throws because of the blackbox.